### PR TITLE
Added Flake8 to Travis

### LINKED ISSUE #478: Integrate flake8 on travis
## Description
As a developer,
I need to ensure that the project has some code checker enabled,
so that it can follow standard code style guidelines

## Mocks
N/A

### Update [Required]
- [x] Modify setup.cfg for flake8 settings
- [x] Add command in .travis.yml to run flake8 on repo
- [ ] Make necessary changes
- [ ] Document the flake8 run in docs.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 2 mentor.

## Estimation
First two task will take approx 30 mins.
third task will take 2-3 hours

### PR DESCRIPTION
**Pull request by kamsuri**
Resolves #478 
I have added Flake8 to travis and also reduced Flake8 warnings from 2874 to 834.
Please review the code.

### CODE REVIEW

**Comment by kamsuri** [comment]
@tapasweni-pathak Can you please guide me a bit on why my tasks are failing and if you can have a look at the code?

File: vms/administrator/tests/test_formFields.py [hunk 1]
```diff
@@ -188,8 +185,8 @@ def test_null_values_in_create_shift(self):
         created_event = create_event_with_details(event)
 
         # create job with values
-        job = ['job', '2017-08-21', '2017-08-21', '',created_event]
-        created_job = create_job_with_details(job)
+        job = ['job', '2017-08-21', '2017-08-21', '', created_event]
+        create_job_with_details(job)
```
Comment (tapasweni-pathak) [inline vms/administrator/tests/test_formFields.py]: we need to assign this to ```created_job```.

File: vms/administrator/tests/test_formFields.py [hunk 2]
```diff
@@ -269,12 +266,12 @@ def test_field_value_retention_for_event(self):
         # verify that event was not edited and that field values are not
         # erased
         event = ['event-name', '2017-08-21', '2017-09-28']
-        created_event = create_event_with_details(event)
+        create_event_with_details(event)
```
Comment (tapasweni-pathak) [inline vms/administrator/tests/test_formFields.py]: ```created_event``` here.

File: vms/administrator/tests/test_formFields.py [hunk 3]
```diff
@@ -310,8 +307,8 @@ def test_field_value_retention_for_job(self):
         # now create job and edit it
         # verify that job was not edited and that field values are not
         # erased
-        job = ['job', '2017-08-21', '2017-08-21', '',created_event]
-        created_job = create_job_with_details(job)
+        job = ['job', '2017-08-21', '2017-08-21', '', created_event]
+        create_job_with_details(job)
```
Comment (tapasweni-pathak) [inline vms/administrator/tests/test_formFields.py]: ```created_job``` here.

File: vms/administrator/tests/test_formFields.py [hunk 4]
```diff
@@ -367,37 +364,37 @@ def test_field_value_retention_for_shift(self):
...
 
         # verify that error message displayed
         self.assertEqual(settings.get_shift_max_volunteer_error(),
-            'Ensure this value is greater than or equal to 1.')
+                         'Ensure this value is greater than or equal to 1.')
 
         # Create shift and try editing it with 0 value
         shift = ['2017-08-21', '09:00', '12:00', '10', created_job]
-        created_shift = create_shift_with_details(shift)
+        create_shift_with_details(shift)
```
Comment (tapasweni-pathak) [inline vms/administrator/tests/test_formFields.py]: ```created_shift``` here.

File: vms/administrator/tests/test_formFields.py [hunk 5]
```diff
@@ -412,7 +409,7 @@ def test_simplify_shift(self):
 
         # Create shift and check job details in edit form
         shift = ['2017-08-21', '09:00', '12:00', '10', created_job]
-        created_shift = create_shift_with_details(shift)
+        create_shift_with_details(shift)
```
Comment (tapasweni-pathak) [inline vms/administrator/tests/test_formFields.py]: ```created_shift``` here.

File: vms/administrator/tests/test_report.py [hunk 6]
```diff
@@ -1,10 +1,8 @@
 # third party
 from selenium import webdriver
-from selenium.common.exceptions import NoSuchElementException
```
Comment (tapasweni-pathak) [inline vms/administrator/tests/test_report.py]: This is not being used in this file?

File: vms/administrator/tests/test_report.py [hunk 7]
```diff
@@ -119,23 +117,23 @@ def test_only_logged_shifts_are_reported(self):
...
+        job = ['Developer', '2017-08-21', '2017-08-30', '', created_event]
         created_job = create_job_with_details(job)
 
         # create shift
         shift = ['2017-08-21', '09:00', '15:00', '10', created_job]
         created_shift = create_shift_with_details(shift)
 
         # shift is assigned to volunteer-one, but hours have not been logged
-        volunteer_shift = register_volunteer_for_shift_utility(created_shift, volunteer)
+        register_volunteer_for_shift_utility(created_shift, volunteer)
```
Comment (tapasweni-pathak) [inline vms/administrator/tests/test_report.py]: Assign this to ```volunteer_shift```.

File: vms/authentication/tests/test_login.py [hunk 8]
```diff
@@ -37,8 +35,8 @@ def setUpClass(cls):
         super(TestAccessControl, cls).setUpClass()
 
     def setUp(self):
-        admin = create_admin()
-        volunteer = create_volunteer()
+        create_admin()
```
Comment (tapasweni-pathak) [inline vms/authentication/tests/test_login.py]: ```setup``` and ```teardown``` will be called, can you please reason the removed of assignment of these two to ```admin``` and ```volunteer```?

File: vms/event/tests/test_shiftSignUp.py [hunk 9]
```diff
@@ -47,17 +46,17 @@ def tearDownClass(cls):
...
         sign_up_page = self.sign_up_page
         sign_up_page.navigate_to_sign_up()
-        self.assertEqual(sign_up_page.get_info_box().text,sign_up_page.no_event_message)
+        self.assertEqual(sign_up_page.get_info_box().text, sign_up_page.no_event_message)
 
     def test_signup_shifts_with_registered_shifts(self):
 
-        created_event = register_event_utility()
-        created_job = register_job_utility()
+        register_event_utility()
```
Comment (tapasweni-pathak) [inline vms/event/tests/test_shiftSignUp.py]: ```created_event``` and ```created_job``` here.

File: vms/shift/tests/test_manageVolunteerShift.py [hunk 10]
```diff
@@ -296,32 +297,32 @@ def test_slots_remaining_in_shift(self):
...
+        self.assertEqual(sign_up_page.get_info_box().text, sign_up_page.no_event_message)
+
     def test_cancel_assigned_shift(self):
         sign_up_page = self.sign_up_page
         manage_shift_page = self.manage_shift_page
         self.manage_shift_page.live_server_url = self.live_server_url
 
         # register volunteers
-        v1 = create_volunteer_with_details(self.volunteer_1)
+        create_volunteer_with_details(self.volunteer_1)
```
Comment (tapasweni-pathak) [inline vms/shift/tests/test_manageVolunteerShift.py]: here ```v1``` and ```s1``` are not being used?

File: requirements.txt [hunk 11]
```diff
@@ -1,8 +1,14 @@
+configparser==3.5.0
```
[the remark this answers is not in the document]
Comment (kamsuri) [reply requirements.txt]: They get installed when flake8 is installed, so should i add it in requirememts.txt ?
image

File: vms/administrator/tests/test_formFields.py (hunk 1 above)
Comment (kamsuri) [reply vms/administrator/tests/test_formFields.py]: I will udate all the warnings in other pull request, these assignment statements got changed when i ran autoflake commands. But i will take care of these in other PR.

Comment (tapasweni-pathak) [reply vms/administrator/tests/test_formFields.py]: Cool!

**Pull request closed without merging**